bids/layout/tests/test_path_building.py

```diff
@@ -46,3 +46,92 @@ def test_insufficient_entities(layout, strict, validate):
     """Check https://github.com/bids-standard/pybids/pull/574#discussion_r366447600."""
     with pytest.raises(ValueError):
         layout.build_path({'subject': '01'}, strict=strict, validate=validate)
+
+
+"""
+The following tests that indexes some datasets from bids examples, 
+parses the files for entities and 
+reconstructs the fullpath of each file by relying on pybids config 
+and compares it what's actually in the dataset
+"""
+
+@pytest.mark.parametrize("scope", ["raw"])
+@pytest.mark.parametrize(
+    "dataset",
+    [
+        ("qmri_irt1"),
+        ("qmri_mese"),
+        ("qmri_mp2rage"),
+        ("qmri_mp2rageme"),
+        ("qmri_mtsat"),
+        ("qmri_sa2rage"),
+        ("qmri_vfa"),
+        ("ds000117"),
+    ],
+)
+def test_path_building_on_examples_with_derivatives(dataset, scope, bids_examples):
+    layout = BIDSLayout(bids_examples / dataset, derivatives=True)
+    files = layout.get(subject=".*", datatype=".*", regex_search = True, scope=scope)
+    for bf in files:
+        entities = bf.get_entities()
+        path = layout.build_path(entities)
+        assert(path==bf.path)
+
+@pytest.mark.parametrize(
+    "dataset",
+    [
+        ("micr_SEM"),
+        ("micr_SPIM"),
+        ("asl001"),
+        ("asl002"),
+        ("asl003"),
+        ("asl004"),
+        ("asl005"),
+        ("pet001"),
+        ("pet002"),
+        ("pet003"),
+        ("pet004"),
+        ("pet005"),
+        ("qmri_megre"),
+        ("qmri_tb1tfl"),
+        ("eeg_cbm"),
+        ("ieeg_filtered_speech"),
+        ("ieeg_visual_multimodal"),
+        ("ds000248"),
+        ("ds001"),
+        ("ds114"),
+    ],
+)
+def test_path_building_on_examples_with_no_derivatives(dataset, bids_examples):
+    layout = BIDSLayout(bids_examples / dataset, derivatives=False)
+    files = layout.get(subject=".*", datatype=".*", regex_search =  True)
+    for bf in files:
+        entities = bf.get_entities()
+        path = layout.build_path(entities)
+        assert(path==bf.path)
+
+@pytest.mark.parametrize("scope", ["raw"])
+@pytest.mark.parametrize(
+    "dataset",
+    [
+        pytest.param(
+            "ds000247",
+            marks=pytest.mark.xfail(strict=True,
+                reason="meg ds folder"
+            ),
+        ),
+        pytest.param(
+            "ds000246",
+            marks=pytest.mark.xfail(strict=True,
+                reason="meg ds folder"
+            ),
+        ),
+    ],
+)
+def test_path_building_on_examples_with_derivatives_meg_ds_folder(dataset, scope, bids_examples):
+    layout = BIDSLayout(bids_examples / dataset, derivatives=True)
+    files = layout.get(subject=".*", datatype=".*", regex_search = True, scope=scope)
+    for bf in files:
+        entities = bf.get_entities()
+        path = layout.build_path(entities)
+        assert(path==bf.path)
```

Comment:
I may be wrong but the way pybids parses the `.ds` meg files makes it hard to then reconstruct the "file" names